src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -71,7 +71,12 @@ def addon_widget_id(addon)
       # @param [Boolean] enable true for filtering beta releases
       def filter_beta_releases(enable)
         self.class.filter_beta = enable
-        @addons = enable ? @all_addons.reject(&:beta_release?) : @all_addons
+        if enable
+          @addons, available_addons = @all_addons.partition(&:registered?)
```

Comment:
I would extend the test to check that the content of the list. You could just add some add-on and just check if it was added or not to the list.

You could do that checking if the call to [Yast::UI::ChangeWidget](https://github.com/gilsonsouza/yast-registration/blob/af91a26a5290066d2d395d3b0591dd2b4e12bc2c/src/lib/registration/ui/addon_selection_base_dialog.rb#L251) call includes that add-on in the last parameter. What's your opinion?

So you could have: 

```ruby
context "when beta versions are not filtered" do
  it "sets the filter as not checked in the UI" do
     # the test you already has
  end

  it "displays beta add-ons" do
    # DISCLAIMER: not tested :)
    # the new test could be something like this:
    expect(Yast::UI).to receive(:ChangeWidget)
      .with(Yast::Term.new(:id, :items), :Value, /name of the beta add-on/) # you can use a regular expression to match with a string
    expect(Yast::UI).to receive(:UserInput).and_return(:next)
    dialog.run
  end
```

Sorry for noticing before about it.